src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { SemVer } from 'semver';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { Product } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { traceWarning } from '../../../platform/logging';
+import { IsCodeSpace } from '../../../platform/common/types';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { EnvironmentType } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { IDataViewerDependencyService } from './types';
+import { IKernel } from '../../../kernels/types';
+import { executeSilently } from '../../../kernels/helpers';
+
+export const minimumSupportedPandaVersion = '0.20.0';
+export const getVersionOfPandasCommand =
+    'import pandas as _VSCODE_pandas;print(_VSCODE_pandas.__version__);del _VSCODE_pandas';
+
+function isVersionOfPandaSupported(version: SemVer) {
+    return version.compare(minimumSupportedPandaVersion) > 0;
+}
+
+/**
+ * Responsible for managing dependencies of a Data Viewer.
+ */
+@injectable()
+export class DataViewerDependencyService implements IDataViewerDependencyService {
+    constructor(
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
+        @inject(IsCodeSpace) private isCodeSpace: boolean
+    ) {}
+
+    private packaging(kernel: IKernel): 'pip' | 'conda' {
+        const envType = kernel.kernelConnectionMetadata.interpreter?.envType;
+        const isConda = envType === EnvironmentType.Conda;
+        return isConda ? 'conda' : 'pip';
+    }
+
+    public async checkAndInstallMissingDependenciesOnEnvironment(): Promise<void> {
+        throw new Error('Not implemented');
+    }
+
+    public async checkAndInstallMissingDependenciesOnKernel(kernel: IKernel): Promise<void> {
+        const pandasVersion = await this.getVersionOfPandas(kernel);
```

Comment:
tomorrow, if someone comes along and modifies the code to call this method instead of the ones with the argument, it's not wrong.
but will break things.
i think the problem is there's no way for someone to know which methods to call and when.
hence the suggestion to have a single method that takes the kernel argument.
as mentioned earlier from the kernel we can get the interpreter in one line.

we can get the kernel for any given resource as well, all you need is the nottebook or notebookuri. i

Reply:
I do like that idea! Thank you! I'll figure it out 👍